Sources/PrimerSDK/Classes/User Interface/Text Fields/PrimerZipCodeFieldView.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  PrimerZipCodeFieldView.swift
+//  PrimerSDK
+//
+//  Created by Carl Eriksson on 10/11/2021.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+public final class PrimerZipCodeFieldView: PrimerTextFieldView {
+    
+    internal var zipCode: String {
+        return textField._text ?? ""
+    }
+    
+    override func xibSetup() {
+        super.xibSetup()
+        
+        textField.keyboardType = .namePhonePad
+        textField.isAccessibilityElement = true
+        textField.accessibilityIdentifier = "zip_code_txt_fld"
+        textField.delegate = self
+        isValid = { text in
+            // todo: look into more sophisticated zip code validation, ascii check for now
+            return text.isTypingValidZipCode
+        }
+    }
+    
+    // todo: refactor into separate functions somewhere
+    public override func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        let positionOriginal = textField.beginningOfDocument
+        let cursorLocation = textField.position(from: positionOriginal, offset: (range.location + NSString(string: string).length))
+        
+        guard let primerTextField = textField as? PrimerTextField else { return true }
+        let currentText = primerTextField._text ?? ""
+        let newText = (currentText as NSString).replacingCharacters(in: range, with: string) as String
+        
+        if newText.isEmpty {
+            delegate?.primerTextFieldView(self, didDetectZipCode: nil)
+        } else {
+            delegate?.primerTextFieldView(self, didDetectZipCode: newText)
```

Comment:
I think I get it now, so it's not that it detected a zip code, it's that the text **could** be a valid zip code, right? If this is the case, why the `isValid()` delegate function isn't used?